packages/flutter/lib/src/widgets/semantics_debugger.dart

```diff
@@ -257,7 +257,13 @@ void _paintMessage(Canvas canvas, SemanticsNode node) {
   canvas.save();
   canvas.clipRect(rect);
   final TextPainter textPainter = new TextPainter()
-    ..text = new TextSpan(style: _messageStyle, text: message)
+    ..text = new TextSpan(
+      style: _messageStyle,
+      text: message,
+    )
+    // We default to LTR because we have to pick _something_ if we're given a string
+    // with no direction, and LTR is what Unicode picked as their default, so...
+    ..textDirection = node.textDirection ?? TextDirection.ltr
```

Comment:
Maybe we should show an error instead?  This code is likely to hide bugs.

Reply:
Turns out this was wrong. _getMessage was returning LTR text with embedded potentially-RTL text. Using node.textDirection here would have not worked in RTL environments.

I've made this code better.